test/bidder_submitter.test.ts

```diff
@@ -0,0 +1,276 @@
+
+// Mock dependencies
+jest.mock('../src/utils/db');
+jest.mock('../src/utils/soroban_helper');
+jest.mock('../src/auction');
+
+jest.mock('@blend-capital/blend-sdk');
+jest.mock('../src/utils/slack_notifier');
+jest.mock('@stellar/stellar-sdk', () => {
+  const actual = jest.requireActual('@stellar/stellar-sdk');
+  return {
+    ...actual,
+    SorobanRpc: {
+      ...actual.SorobanRpc,
+      Server: jest.fn().mockImplementation(() => ({
+        getLatestLedger: jest.fn().mockResolvedValue({ sequence: 999 }),
+      })),
+    },
+  };
+});
+jest.mock('../src/utils/config.js', () => {
+  return {
+    APP_CONFIG: {
+      rpcURL: 'http://localhost:8000/rpc',
+      networkPassphrase: 'Public Global Stellar Network ; September 2015',
+      poolAddress: 'CBP7NO6F7FRDHSOFQBT2L2UWYIZ2PU76JKVRYAQTG3KZSQLYAOKIF2WB',
+      backstopAddress: 'CAO3AGAMZVRMHITL36EJ2VZQWKYRPWMQAPDQD5YEOF3GIF7T44U4JAL3',
+      backstopTokenAddress: 'CAS3FL6TLZKDGGSISDBWGGPXT3NRR4DYTZD7YOD3HMYO6LTJUVGRVEAM',
+      usdcAddress: 'CCW67TSZV3SSS2HXMBQ5JFGCKJNXKZM7UQUWUZPUTHXSTZLEO7SJMI75',
+      blndAddress: 'CD25MNVTZDL4Y3XBCPCJXGXATV5WUHHOWMYFF4YBEGU5FCPGMYTVG5JY',
+      keypair: '',
+      fillers: [],
+    },
+  };
+});
+describe('BidderSubmitter', () => {
+  let bidderSubmitter: BidderSubmitter;
+  let mockDb: jest.Mocked<AuctioneerDatabase>;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    mockDb = inMemoryAuctioneerDb() as jest.Mocked<AuctioneerDatabase>;
+    bidderSubmitter = new BidderSubmitter(mockDb);
+  });
+
+  describe('submit', () => {
+    it('should submit a bid successfully', async () => {
+      const mockLoadAuction = jest.fn().mockResolvedValue({
+        bid: new Map<string, bigint>([['USD', BigInt(123)]]),
+        lot: new Map<string, bigint>([['USD', BigInt(456)]]),
+        block: 500,
+      });
+      const mockSubmitTransaction = jest.fn().mockResolvedValue({
+        ledger: 1000,
+        txHash: 'mock-tx-hash',
+        latestLedgerCloseTime: '2023-01-01T00:00:00Z',
+      });
+
+      (SorobanHelper as jest.MockedClass<typeof SorobanHelper>).mockImplementation(
+        () =>
+          ({
+            loadAuction: mockLoadAuction,
+            submitTransaction: mockSubmitTransaction,
+            network: { rpc: 'mock-rpc', opts: {} },
+          }) as unknown as SorobanHelper
+      );
+
+      (PoolContract as jest.MockedClass<typeof PoolContract>).mockImplementation(
+        () =>
+          ({
+            submit: jest.fn().mockReturnValue('mock-operation'),
+          }) as unknown as PoolContract
+      );
+
+      jest
+        .spyOn(auctionModule, 'calculateBlockFillAndPercent')
+        .mockResolvedValue({ fillBlock: 1000, fillPercent: 0.5 });
+      jest.spyOn(auctionModule, 'scaleAuction').mockReturnValue({
+        bid: new Map<string, bigint>([['USD', BigInt(12)]]),
+        lot: new Map<string, bigint>([['USD', BigInt(34)]]),
+        block: 500,
+      });
```

Comment:
We should fix this usage of Jest to make the tests much more readable and maintainable

I would expect

```js
import { myDepFn } from './path'

jest.mock('./path');

describe("mySystemUnderTest") {
    let mockedMyDepFn = myDepFn as jest.MockedFunction<typeof myDepFn>;
    
    it("is mocked") {
        mockedMyDepFn.mockReturnValue(true)
    }
}
```

Reply:
(this goes for all usages of spyOn to mock functions within the auctionModule or similar)